src/test/java/com/redhat/red/build/finder/MockKojiClientSession.java

```diff
@@ -0,0 +1,127 @@
+package com.redhat.red.build.finder;
+
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.FilenameUtils;
+import org.commonjava.rwx.api.RWXMapper;
+import org.commonjava.rwx.core.Registry;
+
+import com.redhat.red.build.koji.KojiClientException;
+import com.redhat.red.build.koji.model.generated.Model_Registry;
+import com.redhat.red.build.koji.model.xmlrpc.KojiArchiveInfo;
+import com.redhat.red.build.koji.model.xmlrpc.KojiArchiveQuery;
+import com.redhat.red.build.koji.model.xmlrpc.KojiArchiveType;
+import com.redhat.red.build.koji.model.xmlrpc.KojiBuildInfo;
+import com.redhat.red.build.koji.model.xmlrpc.KojiTagInfo;
+import com.redhat.red.build.koji.model.xmlrpc.KojiTaskInfo;
+import com.redhat.red.build.koji.model.xmlrpc.KojiTaskRequest;
+import com.redhat.red.build.koji.model.xmlrpc.messages.GetArchiveTypesResponse;
+import com.redhat.red.build.koji.model.xmlrpc.messages.GetBuildResponse;
+import com.redhat.red.build.koji.model.xmlrpc.messages.GetTaskRequestResponse;
+import com.redhat.red.build.koji.model.xmlrpc.messages.GetTaskResponse;
+import com.redhat.red.build.koji.model.xmlrpc.messages.ListArchivesResponse;
+import com.redhat.red.build.koji.model.xmlrpc.messages.ListTagsResponse;
+
+public class MockKojiClientSession implements ClientSession {
+    private static final String MESSAGES_BASE = "messages";
+
+    private RWXMapper rwxMapper;
+
+    private String resourcePath;
+
+    public MockKojiClientSession() {
+        this(MESSAGES_BASE);
+    }
+
+    public MockKojiClientSession(String resourcePath) {
+        Registry.setInstance(new Model_Registry());
+        this.rwxMapper = new RWXMapper();
+        this.resourcePath = resourcePath;
+    }
+
+     private String readResource(String resourceFile) throws IOException {
+        String resource = FilenameUtils.separatorsToUnix(FilenameUtils.concat(resourcePath, resourceFile));
+        File file = TestUtils.loadFile(resource);
+        String fileContents = FileUtils.readFileToString(file, "UTF-8");
+        return fileContents;
+    }
+
+    private <T> T parseCapturedMessage(Class<T> type, String filename) throws Exception {
+        String source = readResource(filename);
+        T parsed = rwxMapper.parse(new ByteArrayInputStream(source.getBytes()), type);
+        return parsed;
+    }
+
+    @Override
+    public List<KojiArchiveInfo> listArchives(KojiArchiveQuery query) throws KojiClientException {
+        try {
+            ListArchivesResponse response = parseCapturedMessage(ListArchivesResponse.class, "listArchives-response.xml");
+            return response.getArchives();
```

Comment:
These look like hard-coded responses? How will we handle the scenario where we want to have different responses for different tests?

Reply:
The way it's written now, each test could get it's own resource subdirectory and then pass that to the constructor, as the filenames are hardcoded.

The other way to do it is to allow the file names to be changed, but this was a more complicated change whereas the directory change was more simple to make.

Reply:
Ah ok, thanks the directory sounds fine for now!